### PR TITLE
Portal nexus menu entry swapper

### PR DESCRIPTION
**Pull request by kyleshepherd**
Attempt at implementing the suggestion from #11267.

This has been done by implementing a config setting in the Menu Entry Swapper UI which, if activated, will swap any set Left Click teleport for the Teleport Menu option.

Not sure if this is the most efficient way to do it so happy for feedback!

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 1]
```diff
@@ -331,6 +331,17 @@ default boolean swapJewelleryBox()
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "swapPortalNexus",
+		name = "Portal Nexus",
+		description =  "Swap Left Click option with Teleport Menu",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java:337]:
```suggestion
		description =  "Swap Teleport options with Teleport Menu on the Portal Nexus",

```

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 2]
```diff
@@ -304,6 +304,42 @@ void setupSwaps()
...
+		swap("waterbirth island", "teleport menu", config::swapPortalNexus);
+		swap("barrows", "teleport menu", config::swapPortalNexus);
+		swap("carrallangar", "teleport menu", config::swapPortalNexus);
+		swap("fishing guild", "teleport menu", config::swapPortalNexus);
+		swap("catherby", "teleport menu", config::swapPortalNexus);
+		swap("annakarl", "teleport menu", config::swapPortalNexus);
+		swap("ape atoll dungeon", "teleport menu", config::swapPortalNexus);
+		swap("ghorrok", "teleport menu", config::swapPortalNexus);
+		swap("troll stronghold", "teleport menu", config::swapPortalNexus);
+		swap("weiss", "teleport menu", config::swapPortalNexus);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:339]: This should be formatted more like the NPC contact swaps. e.g.

```java
Arrays.asList(
	"grand exchange", "varrock", ...
).forEach(teleport -> swap(teleport, "teleport menu", config::swapPortalNexus));
```

In hindsight, I'm not entirely sure why the jewellery box teleport options weren't done the same way. That's optimization to worry about later

**Comment by kyleshepherd** [comment]
Thanks for the feedback @Nightfirecat! I changed it to how you suggested, much cleaner!

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 3]
```diff
@@ -304,6 +304,15 @@ void setupSwaps()
 		swap("teleport menu", "draynor village", config::swapJewelleryBox);
 		swap("teleport menu", "al kharid", config::swapJewelleryBox);
 
+		Arrays.asList(
+				"grand exchange", "varrock", "lumbridge grave'", "draynor manor", "battlefront", "mind altar",
+				"lumbridge", "falador", "salve graveyard", "camelot", "seers' village", "fenken' castle",
+				"east ardougne", "watchtower", "yanille", "senntisten", "west ardougne", "marim", "harmony island",
+				"kharyll", "lunar isle", "kourend castle", "the forgotten cemetery", "waterbirth island", "barrows",
+				"carrallangar", "fishing guild", "catherby", "annakarl", "ape atoll dungeon", "ghorrok",
+				"troll stronghold", "weiss"
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:313]: This doesn't seem to exactly match the teleport options available for the portal nexus. The following are the teleport options the nexus offers:

```
annakarl
ape atoll dungeon
ardougne
barrows
battlefront
camelot
carrallangar
catherby
cemetery
draynor manor
falador
fenkenstrain's castle
fishing guild
ghorrock
grand exchange
great kourend
harmony island
kharyrll
lumbridge
lumbridge graveyard
lunar isle
marim
mind altar
salve graveyard
seers' village
senntisten
troll stronghold
varrock
watchtower
waterbirth island
weiss
west ardougne
yanille
```

Notably these are the entries you have which don't match portal nexus options:
* east ardougne
* the forgotten cemetery
* fenken' castle
* kourend castle
* lumbridge grave

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 4]
```diff
@@ -304,6 +304,15 @@ void setupSwaps()
...
+			).forEach(location -> swap(location, "teleport menu", config::swapPortalNexus));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:314]: Reduce this indentation by one level.
```suggestion
			"grand exchange", "varrock", "lumbridge grave'", "draynor manor", "battlefront", "mind altar",
			"lumbridge", "falador", "salve graveyard", "camelot", "seers' village", "fenken' castle",
			"east ardougne", "watchtower", "yanille", "senntisten", "west ardougne", "marim", "harmony island",
			"kharyll", "lunar isle", "kourend castle", "the forgotten cemetery", "waterbirth island", "barrows",
			"carrallangar", "fishing guild", "catherby", "annakarl", "ape atoll dungeon", "ghorrok",
			"troll stronghold", "weiss"
		).forEach(location -> swap(location, "teleport menu", config::swapPortalNexus));
```

**Review by Nightfirecat — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java (hunk 3 above)
Comment (kyleshepherd) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:313]: Ah, I was following the wiki's listing which I think has some of them shortened for brevity, I'll get this changed!

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 5]
```diff
@@ -304,6 +304,14 @@ void setupSwaps()
 		swap("teleport menu", "draynor village", config::swapJewelleryBox);
 		swap("teleport menu", "al kharid", config::swapJewelleryBox);
 
+		Arrays.asList(
+			"annakarl", "ape atoll dungeon", "ardougne", "barrows", "battlefront", "camelot", "carrallangar",
+			"catherby", "cemetery", "draynor manor", "falador", "fenkenstrain's castle", "fishing guild", "ghorrock",
+			"grand exchange", "great kourend", "harmony island", "kharyrll", "lumbridge", "lumbridge graveyard",
+			"lunar isle", "marim", "mind altar", "salve graveyard", "seers' village", "senntisten", "troll stronghold",
+			"varrock", "watchtower", "waterbirth island", "weiss", "west ardougne", "yanille"
+		).forEach(location -> swap(location, "teleport menu", config::swapPortalNexus));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:313]: I'll commit this via GitHub's UI
```suggestion
		).forEach(location -> swap(location, "portal nexus", "teleport menu", config::swapPortalNexus));
```

Comment (kyleshepherd) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Ah good catch, ty

**Pull request merged**